src/CreateBoardForm.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { Component, FormEvent } from 'react'
+import { remove, shuffle } from 'lodash'
+import { Checkbox } from './Checkbox'
+
+export class CreateBoardForm extends Component<{}, { chosenCharacters: string[], showBoard: boolean, board: string[] }> {
+  baseCharacters: string[] = [
+    "Arya Stark",
+    "Sansa Stark",
+    "Bran Stark",
+    "Jon Snow",
+    "Theon Greyjoy",
+    "Yara Greyjoy",
+    "Euron Greyjoy",
+    "Cersei Lannister",
+    "Jaime Lannister",
+    "Tyrion Lannister",
+    "Sam Tarly",
+    "Tormund Giantsbane",
+    "Brienne of Tarth",
+    "Gendry",
+    "Podrick",
+    "Danaerys Targaryen",
+    "Jorah Mormont",
+    "Davos Seaworth",
+    "Lyanna Mormont",
+    "The Night King"
+  ]
+
+  optionalCharacters: string[] = [
+    "Lord Varys",
+    "Missandei",
+    "Grey Worm",
+    "The Hound",
+    "Beric Dondarrion",
+    "Bronn",
+    "Melisandre",
+    "Edd Tollett",
+    "Gilly",
+    "Little Sam",
+    "Drogon",
+    "Rhaegal"
+  ]
+
+  state = {
+    chosenCharacters: [] as string[],
+    showBoard: false,
+    board: [] as string[]
+  }
+
+  handleCheckboxChange = (event: FormEvent): void => {
+    let { chosenCharacters } = this.state
+    const target = event.target as HTMLInputElement
+
+    if (target.checked) {
+      chosenCharacters.push(target.name)
```

Comment:
I imagine this is how you do state changes in general at Tandem? If so I don't think it'll necessarily cause issues, but if it were me I'd always pull `const`s out of state and perform only immutable operations on state. You're mutating state here, even if it's only temporarily.

Reply:
Oh I guess I thought destructuring also made a shallow copy but I guess not. Can I do `{ [...chosenCharacters] } = this.state`?

Reply:
Or something like that?

Reply:
you can make a shallow copy of the array to avoid mutating state, yeah. Thats all I'm suggesting:
```js
if (target.checked) {
    this.setState({chosenCharacters: [...chosenCharacters, target.name]});
} else {
    this.setState({/* etc */})
}
```